src/operators/Remap/avtRemapFilter.C

```diff
@@ -131,447 +138,476 @@ avtRemapFilter::SetAtts(const AttributeGroup *a)
+                vtkVisItClipper* clipperBack = vtkVisItClipper::New();
+
+                // Left plane
+                clipperLeft->SetInputData(in_ds);
+                clipperLeft->SetClipFunction(funcsArrayX[i]);
+                clipperLeft->SetInsideOut(false);
+                clipperLeft->SetRemoveWholeCells(false);
+                last = clipperLeft;
+
+                // Right plane
+                clipperRight->SetInputData(in_ds);
+                clipperRight->SetClipFunction(funcsArrayX[i+1]);
+                clipperRight->SetInsideOut(true);
+                clipperRight->SetRemoveWholeCells(false);
+                clipperRight->SetInputConnection(last->GetOutputPort());
+                last = clipperRight;
+
+                // Top plane
+                clipperTop->SetInputData(in_ds);
+                clipperTop->SetClipFunction(funcsArrayY[j]);
+                clipperTop->SetInsideOut(false);
+                clipperTop->SetRemoveWholeCells(false);
+                clipperTop->SetInputConnection(last->GetOutputPort());
+                last = clipperTop;
+
+                // Bottom plane
+                clipperBottom->SetInputData(in_ds);
+                clipperBottom->SetClipFunction(funcsArrayY[j+1]);
+                clipperBottom->SetInsideOut(true);
+                clipperBottom->SetRemoveWholeCells(false);
+                clipperBottom->SetInputConnection(last->GetOutputPort());
+                last = clipperBottom;
+
+                if (is3D)
                 {
-                    value += myVariable->GetComponent(tuple, 0) *
-                        subCellVolumes->GetComponent(tuple, 0);
+                    // Front plane
+                    clipperFront->SetInputData(in_ds);
+                    clipperFront->SetClipFunction(funcsArrayZ[k]);
+                    clipperFront->SetInsideOut(false);
+                    clipperFront->SetRemoveWholeCells(false);
+                    clipperFront->SetInputConnection(last->GetOutputPort());
+                    last = clipperFront;
+
+                    // Back plane
+                    clipperBack->SetInputData(in_ds);
+                    clipperBack->SetClipFunction(funcsArrayZ[k+1]);
+                    clipperBack->SetInsideOut(true);
+                    clipperBack->SetRemoveWholeCells(false);
+                    clipperBack->SetInputConnection(last->GetOutputPort());
+                    last = clipperBack;
                 }
-                value /= rCellVolume;
-                vars[vdx]->SetComponent(rCell, 0, value);
-            }
-            else if (varType == RemapAttributes::extrinsic) // like mass
-            {
-                for (vtkIdType tuple = 0;
-                     tuple < myVariable->GetNumberOfTuples(); tuple++)
+
+                last->Update();
+                vtkUnstructuredGrid* ug = last->GetOutput();
+                debug5 << "Cell " << rCell << " has been clipped." << std::endl;
```

Comment:
Please delete your clippers when you are done with them.

Reply:
Done.